source/convective_grad/main.cpp

```diff
@@ -228,6 +229,39 @@ void main_main()
                 ga(i,j,k,1) = eos_state.p * chi_T / (eos_state.gam1 * eos_state.rho * eos_state.T * eos_state.cv);
 
 
+                // del_ledoux = del_ad + B, where B is the composition term
+                // We calculate it like MESA, Paxton+ 2013 Equation 8
+
+                Real lnP1 = log(P(i,j,k)); // true pressure at this grid point
+                Real lnP2 = 0.0; // true pressure at the next grid point
+
+                if (ndims == 1) {
+                    // x is the vertical
+                    lnP2 = log(P(i+1,j,k)); 
+                    for (int n = 0; n < NumSpec; ++n) {
+                        eos_state.xn[n] = fab(i+1,j,k,spec_comp+n); // 
+                    }
```

Comment:
the problem here is that the species don't have ghost cells -- I never fill them, so you will go out of bounds here (compile with DEBUG=TRUE and you should see that).

Reply:
Ok. If it's only an issue at the uppermost grid point, we can just set del_ledoux = del_ad there.

Reply:
I'll fix it properly, the same way I handled dlogT/dlogP